### PR TITLE
MDCT-1857 pt. 2 --add frontend components for form template api

### PR DESCRIPTION
**Pull request by gmrabian**
## Description
Partial [MDCT-1857](https://qmacbis.atlassian.net/browse/MDCT-1857) Adds frontend api components and writes form to db when adding a program

Follows [PR 3804](https://github.com/CMSgov/mdct-mcr/pull/3804)

### How to test
- `./dev local`
- State user -> dashboard -> create program
- DynamoDb admin -> see report and form template, ids should match
- When you edit a report's details from the dashboard the form template id does not change

### Changed Dependencies
Need root .env addition if you haven't already
`FORM_TEMPLATE_TABLE_NAME=local-mcpar-form-templates`

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Review by braxex — APPROVED** [review]
This all looks good to me.

File: services/ui-src/src/forms/mcpar/mcpar.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "MCPAR",
   "basePath": "/mcpar",
-  "version": "0.0.0",
+  "version": "MCPAR_Sep-8-2022",
```
Comment (braxex) [inline services/ui-src/src/forms/mcpar/mcpar.json:4]: Nit: My personal preference is for a numeric-only date and I don't think design will ever see this, so I think we should go for that. I'm not the only person with opinions though so if everyone else wants to go with something like this instead that's cool.

Comment (gmrabian) [reply services/ui-src/src/forms/mcpar/mcpar.json:4]: 🤷 sure. Doesn't matter to me

Comment (gmrabian) [inline services/ui-src/src/forms/mcpar/mcpar.json:4]:
```suggestion
  "version": "MCPAR_2022-09-08", // yyyy-mm-dd
```

**Pull request merged**